### ISSUE TITLE
Mac: notarizing requirement coming with Catalina ( macOS 10.15)

### ISSUE THREAD

**Issue by nofishonfriday** [reaper-oss/sws#1194]
Just reading into this currently, is SWS affected by it?
https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution

>Beginning in macOS 10.15, notarization is required by default for all software.

**Comment by cfillion** [comment]
I don't think this applies to random *.dylib but it mentions "Disk images (UDIF format)" so maybe the .dmg could be "notarized" if we wanted to(?). The dylib and the dmg are not even code signed right now. However...

> Notarization requires Xcode 10 or later. Building a new app for notarization requires macOS 10.13.6 or later. Stapling an app requires macOS 10.12 or later.

Xcode 10 means no 32-bit SWS.

> Link against the macOS 10.9 or later SDK.

Well, at least this would allow fully adopting C++11...

**Comment by jazzido** [comment]
Hi,

Just opened Reaper after upgrading to 10.15 Catalina, and it seems that notarization does apply to dylibs :/

<img width="532" alt="Screen Shot 2019-10-18 at 12 12 59 PM" src="https://user-images.githubusercontent.com/27584/67106368-ddd07b80-f1a0-11e9-9294-b01174390656.png">

**Comment by nofishonfriday** [comment]
For now the workaround posted by Justin may help? (not on Catalina myself..)
https://forum.cockos.com/showpost.php?p=2192445&postcount=10

**Comment by spblat** [comment]
The workaround linked above worked for me.

**Comment by kenokenobingo** [comment]
I tried to freshly install both Reaper and SWS on macOS Catalina, but it’s not working out of the box, using the installer.

<img width="662" alt="SWS on macOS Catalina" src="https://user-images.githubusercontent.com/9639767/68366333-576de080-0133-11ea-9f56-f8f9cbad337d.png">

**Comment by CD20202020** [comment]
> I tried to freshly install both Reaper and SWS on macOS Catalina, but it’s not working out of the box, using the installer.
> 
> <img alt="SWS on macOS Catalina" width="662" src="https://user-images.githubusercontent.com/9639767/68366333-576de080-0133-11ea-9f56-f8f9cbad337d.png">
> 
> EDIT: @nofishonfriday’s workaround is working, but it’s necessary to use the user library instead of /Library for `reaper_sws_extension.dylib`.

Hi,
Want to install SWS to use with Reaper v6.11/64 on macOS Catalina. Have downloaded @nofishonfriday’s workaround.
Where does it go? Help with the last bit of installing it would be great. 
Thanks.

**Comment by CD20202020** [comment]
<img width="1152" alt="Screenshot 2020-06-24 at 14 16 16" src="https://user-images.githubusercontent.com/67374163/85563805-a0709780-b625-11ea-8bfe-5e1c6ebc0ada.png">

**Comment by cfillion** [comment]
Don't rename the .dylib file. `xattr` is a command that should be run using Terminal.app. Alternatively, just allow SWS to be run in System Preferences  > Security after macOS complains when launching REAPER.

**Comment by CD20202020** [comment]
@cfillion Thank you! Went the System Preferences route, granted permission and SWS actions now appearing in Reaper.
